src/Data/Arrays.php

```diff
@@ -260,33 +260,49 @@ public static function multiToAssoc(array $arrays)
     /**
      * Converted a multi-dimensional associative array with `dot`.
      *
-     * @param array $arrays Arrays.
+     * @param array $arrays      Arrays.
+     * @param bool  $assocOutput Switch to output assoc arrays.
      *
      * @since 3.0.0
      *
-     * @return bool
+     * @return array
      */
-    public static function dot(array $arrays)
+    public static function dot(array $arrays, bool $assocOutput = false)
     {
-        return self::multiToAssocWithSpecificOpr($arrays, '.');
+        return self::multiToAssocWithSpecificOpr($arrays, '.', $assocOutput);
     }
-
     /**
      * Converted a multi-dimensional associative array with `operator`.
      *
-     * @param array  $arrays Arrays.
-     * @param string $opr    Operator.
+     * @param array  $arrays      Arrays.
+     * @param string $opr         Operator.
+     * @param bool   $assocOutput Switch to output assoc arrays.
+     * @param string $_key        the previous key of the object
      *
      * @since 3.0.0
      *
      * @return array
      */
-    public static function multiToAssocWithSpecificOpr(array $arrays, $opr = null)
+    public static function multiToAssocWithSpecificOpr(array $arrays, $opr = null, bool $assocOutput = false, string $_key = null)
     {
         $results = [];
         foreach ($arrays as $key => $value) {
+            $key = ($assocOutput === true ? $_key.$key : $key);
             if (self::isReallyArray($value) === true) {
-                $results = array_merge($results, self::multiToAssocWithSpecificOpr($value, $key.$opr));
+                $assocKey = ($assocOutput === true ? $opr : $key.$opr);
+                $results = array_merge(
+                    $results,
+                    self::multiToAssocWithSpecificOpr(
+                        $value,
+                        $assocKey,
+                        $assocOutput,
+                        $key.$opr
+                    )
+                );
+            } elseif ($assocOutput === true) {
+                // gets pull off the last dot that is added
+                $_key = preg_replace('/(\.$)/', '', $_key);
```

Comment:
It should not be dot specific.

Reply:
You may consider Opr variable

Reply:
Hi, thank you for that, I will change that now.